Xamarin.Essentials/Types/Brightness.shared.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+#if WINDOWS_UWP
+using UwpBrightness = Windows.Graphics.Display.BrightnessOverride;
+#endif
+
+namespace Xamarin.Essentials
+{
+    [Preserve(AllMembers = true)]
+    public readonly partial struct BrightnessOverride : IDisposable, IEquatable<BrightnessOverride>
+    {
+        internal BrightnessOverride(Brightness oldBrightness, Brightness appliedBrightness)
+        {
+            OldBrightness = oldBrightness;
+            AppliedBrightness = appliedBrightness;
+#if WINDOWS_UWP
+            @override = null;
+#endif
+        }
+
+#if WINDOWS_UWP
+        readonly UwpBrightness @override;
+
+        public BrightnessOverride(Brightness oldBrightness, Brightness newBrightness, UwpBrightness @override)
+            : this(oldBrightness, newBrightness)
+        {
+            this.@override = @override;
+        }
+
+        public void Dispose()
+        {
+            @override.StopOverride();
+        }
+
+        public bool Equals(BrightnessOverride other)
+        {
+            return OldBrightness.Equals(other.OldBrightness) && AppliedBrightness.Equals(other.AppliedBrightness) && @override.Equals(other.@override);
+        }
+#endif
+
+        public Brightness OldBrightness { get; }
+
+        public Brightness AppliedBrightness { get; }
+
+#if !WINDOWS_UWP
+        public void Dispose()
+        {
+            AppInfo.SetBrightness(OldBrightness);
+        }
+#endif
+
+#if !WINDOWS_UWP
+        public bool Equals(BrightnessOverride other)
+        {
+            return OldBrightness.Equals(other.OldBrightness) && AppliedBrightness.Equals(other.AppliedBrightness);
+        }
+#endif
+    }
+
+    [Preserve(AllMembers = true)]
+    public readonly struct Brightness
```

Comment:
Why was a class created to describe a single property?

Reply:
Check all other instances of sensor readings, such as barometer.